packages/core/getUserAvailability.ts

```diff
@@ -413,9 +416,54 @@ const _getUserAvailability = async function getUsersWorkingHoursLifeTheUniverseA
     end: dayjs(busy.end),
   }));
 
-  const dateRangesInWhichUserIsAvailable = subtract(dateRanges, formattedBusyTimes);
+  let dateRangesInWhichUserIsAvailable = subtract(dateRanges, formattedBusyTimes);
+
+  let dateRangesInWhichUserIsAvailableWithoutOOO = subtract(oooExcludedDateRanges, formattedBusyTimes);
+
+  if (considerOnlyFirstSlot === true && dateRangesInWhichUserIsAvailableWithoutOOO.length > 0) {
+    // Get the date range of the whole day to get the initial time of the day.
+    const { dateRanges: wholeDayDateRange } = buildDateRanges({
+      dateFrom: dayjs(dateFrom).startOf("date"),
+      dateTo: dayjs(dateTo).endOf("date"),
+      availability,
+      timeZone,
+      travelSchedules: isDefaultSchedule
+        ? user.travelSchedules.map((schedule) => {
+            return {
+              startDate: dayjs(schedule.startDate),
+              endDate: schedule.endDate ? dayjs(schedule.endDate) : undefined,
+              timeZone: schedule.timeZone,
+            };
+          })
+        : [],
+      outOfOffice: datesOutOfOffice,
+    });
+
+    const wholeDayDateRangeExcludingBusyTimes = subtract(wholeDayDateRange, formattedBusyTimes);
```

Comment:
Aren't we doing the same thing in line 397? You should take this out of the if and conditionally fetch dateRanges based on considerOnlyFirstSlot

You can change the way we get dateRanges in line 397 like this:
```suggestion
    const { dateRanges: wholeDayDateRange } = buildDateRanges({
      dateFrom: considerOnlyFirstSlot ? dayjs(dateFrom).startOf("date") : dateFrom,
      dateTo: considerOnlyFirstSlot ? dayjs(dateTo).endOf("date") : dateTo,
      availability,
      timeZone,
      travelSchedules: isDefaultSchedule
        ? user.travelSchedules.map((schedule) => {
            return {
              startDate: dayjs(schedule.startDate),
              endDate: schedule.endDate ? dayjs(schedule.endDate) : undefined,
              timeZone: schedule.timeZone,
            };
          })
        : [],
      outOfOffice: datesOutOfOffice,
    });

    const wholeDayDateRangeExcludingBusyTimes = subtract(wholeDayDateRange, formattedBusyTimes);
```

Reply:
Or maybe not 🤔. I think my suggestion is wrong, ignore it for now

Reply:
Thanks @Amit91848 !

The reason we have to do this two times, is because the first `buildDateRanges` we are fetching the date range for `dateFrom` to `dateTo` which is basically the user selected time frame.

So, for example the user selected time frame of 9:00 - 9:10.

So, the first call will return a dateRange of -
```
[
    {
        start: <date-and-9:00-timestamp>
        end: <date-and-9:10-timestamp>
    }
]
```

And the second call will return something like this -
```
[
  {
     start: <date-and-first-available-time-timestamp>
     end: <date-and-last-available-time-timestamp>
  }
]
```

And if you see the code we will require both the data. Please let me know if you need a more elaborate explanation.
I can explain it. 👍

Reply:
> Or maybe not 🤔. I think my suggestion is wrong, ignore it for now

Haha, I was just writing it. No issues. 👍

Reply:
I know its a little less efficient as we have to call it twice. But I didn't find any other way around.